### PR TITLE
BUG Numpy dev version has different allclose

### PR DESCRIPTION
**Pull request by mhvk**
This PR partially addresses #3584, that with the numpy development version there are many test errors because the default for `atol` in `np.allclose` and `np.testing.assert_allclose` has changed. Almost all problems were in coordinates (hence, cc @eteq) and cosmology (hence, cc @aconley), where comparisons were often done on quantities rather than on values. Note that the changes are not necessarily complete -- but at least the tests pass.

Of course, it is *logical* to compare on quantities, not values (except perhaps in `units` itself), so I changed the tests to use `Quantity` aware versions where there were problems, adding a new `quantity_allclose` to join the existing `assert_quantity_allclose` in the process (hence, cc @astrofrog).

### CODE REVIEW

**Comment by aconley** [comment]
I'm confused -- why were the non-quantity tests changed to use the unit aware versions.
Example: all of the Omega's are not quantities, so what does changing

    assert np.allclose(cosmo.Om0, 0.27)

to

    assert allclose(csomo.Om0, 0.27)

actually do for anyone?

**Comment by mhvk** [comment]
@aconley - that's what one gets from automating the conversion using `replace-string`...  For dimensionless quantities, there indeed was no problem to start with. Anyway, in the end it of course does exactly the same thing; it just seemed easier to change all tests than change just those where units were used when they shouldn't have.  Feel free to submit a PR that either undoes it or only changes the ones that actually cause problems.

**Comment by aconley** [comment]
Meh -- if it's for convenience, that's a good enough reason.  I just wanted to understand if there was some more fundamental issue I should know about with np.allclose.

**Comment by mhvk** [comment]
@eteq - could you also have a look at this? I think it would be good to get it in, so that at least most of the test failures against numpy-dev disappear.

**Comment by mhvk** [comment]
@astrofrog, @eteq - could you have a look at this?  I very much prefer not to have to rebase this and it touches a lot of files, so is risky.

File: astropy/coordinates/tests/test_earth.py [hunk 1]
```diff
@@ -75,19 +74,19 @@ def test_gd2gc():
...
     location = EarthLocation.from_geodetic(e, p, h, ellipsoid='GRS80')
-    xyz = location.to_geocentric()
+    xyz = tuple(v.to(u.m) for v in location.to_geocentric())
     vvd(xyz[0], -5599000.5577260984, 1e-7, "eraGd2gc", "0/2", status)
     vvd(xyz[1], 233011.6722356703, 1e-7, "eraGd2gc", "1/2", status)
     vvd(xyz[2], -3040909.4706095476, 1e-7, "eraGd2gc", "2/2", status)
 
     location = EarthLocation.from_geodetic(e, p, h, ellipsoid='WGS72')
-    xyz = location.to_geocentric()
+    xyz = tuple(v.to(u.m) for v in location.to_geocentric())
```
Comment (astrofrog) [inline astropy/coordinates/tests/test_earth.py]: This change doesn't seem to be related to the allclose changes - can you clarify?

Comment (mhvk) [inline astropy/coordinates/tests/test_earth.py]: `allclose` is used in `vvd`. Before, I relied on values from `to_geocentric` being already in `m` and the unit being ignored; now it explicitly converts to `m` (we might want to change the default, who knows...) and in `vvd` attaches that unit to the comparison (which is inserted using cut & paste input, and which I'd like not to touch too much).

**Comment by astrofrog** [comment]
Thinking about this some more, I think that given the mix of quantity/non-quantity values in coordinates and cosmology, it actually makes sense to use this one always, because it is actually more strict in some cases:

```
In [9]: np.testing.assert_allclose(3 * u.deg, 3)

In [10]: assert_quantity_allclose(3 * u.deg, 3)
...
UnitsError: Units for 'desired' () and 'actual' (deg) are not convertible
```

**Comment by astrofrog** [comment]
I will go ahead and merge this since it has been open for a couple of weeks and the risk of rebase is high.
